src/main/java/org/jenkinsci/plugins/casc/ObsoleteConfigurationMonitor.java

```diff
@@ -0,0 +1,59 @@
+package org.jenkinsci.plugins.casc;
+
+import hudson.Extension;
+import hudson.model.AdministrativeMonitor;
+import hudson.util.VersionNumber;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.casc.model.CNode;
+import org.jenkinsci.plugins.casc.model.Source;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+@Extension
+public class ObsoleteConfigurationMonitor extends AdministrativeMonitor {
+
+    private final List<Error> errors = new CopyOnWriteArrayList<>();
+
+    @Override
+    public boolean isActivated() {
+        return !errors.isEmpty();
+    }
+
+    public List<Error> getErrors() {
+        return errors;
+    }
+
+
+    public void reset() {
+        errors.clear();
+    }
+
+    public void record(CNode node, String error) {
+        errors.add(new Error(node.getSource(), error));
+    }
+
+    public static ObsoleteConfigurationMonitor get() {
+        return AdministrativeMonitor.all().get(ObsoleteConfigurationMonitor.class);
+    }
+
+    public static class Error {
+        public final Source source;
+        public final String message;
+
+        public Error(Source source, String message) {
+            this.source = source;
+            this.message = message;
+        }
+    }
+
+    public String getCss() {
+        if (Jenkins.getVersion().isNewerThan(new VersionNumber("2.103"))) {
+            return "alert alert-warning";
+        }
+        return "warning";
```

Comment:
I don't recommend you to use hard-coded CSS styles in the business logic. The recommended approach is explained here https://jenkins.io/blog/2018/01/21/overhaul-of-manage-jenkins-page/#for-plugin-developers

Reply:
And you can see other implementations https://github.com/jenkinsci/ssh-slaves-plugin/pull/70/files

Reply:
I've read this doc, but selected the simplest solution. a DataBound class is by design dedicated to web UI rendering, so I don't consider it's a mistake to write some rendering logic in plain java that compiler can help me write correctly _vs_ detecting mistakes in made in jelly code at runtime :P

Reply:
Up to you. In any case, my suggestion was already shared with you and the community. Thanks.

Reply:
I think it will be fine once the method is restricted.

Jelly is the same business logic as Java, so I do not see much difference from this PoV. Nicolas followed the recommendations in developer guidelines, but just implemented them in Java instead of Jelly.

Reply:
I agree with @recena the backend should never produce frontend specifc code nor css attributes that is simply really bad SoC.

Reply:
The entire Jelly is converted to Html in the backend 🤷‍♂️

Reply:
You can say the same as filling select box you do that in Java not in jelly 🤷‍♂️

Reply:
Stapler can't compare to REST backend + JS frontend like modern application use nowadays. Most of the code written in a DataBound class is 100% dedicated to front view, and I dislike putting more than trivial and pre-computed properties inside jelly views, as those are a pain to maintain.

Reply:
@oleg-nenashev In that case, we only need to copy and paste the solution. In fact, this solution is being used extensively in Jenkins core and plugins. IMHO, It is positive in term of maintenance to follow these patterns. And avoid exceptions with no needed.

Reply:
I am a frontend dev and this merge is simply pilling up garbage with the excuse of using outdated tech stack when there had been an easy to implement best pattern. The testing  excuse is simply wrong, what benefit is there to test frontend code in the backend? ...anyway I will stop riding the dead horse since this is merged.

Reply:
you still think as "backend vs frontend", which isn't a correct mindset when considering Stapler design.